### PR TITLE
structure deployment/environment_min.yml

### PR DESCRIPTION
**Pull request by basnijholt**
To make it more explicit what gets installed for which reason.

### CODE REVIEW

File: deployment/environment_min.yml [hunk 1]
```diff
@@ -8,38 +8,38 @@ name: py37
...
+  # qms dependencies
+  - deepdish
+  - fenics
+  - kwant
   - mkl
-  - mkl-service
-  - mpi4py
-  - nb_conda
-  - numpy
-  - numpy-stl
```
Comment (basnijholt) [inline deployment/environment_min.yml]: this isn't needed anymore

File: deployment/environment_min.yml [hunk 2]
```diff
@@ -8,38 +8,38 @@ name: py37
...
-  - petsc
```
Comment (basnijholt) [inline deployment/environment_min.yml]: this will be installed when installing `fenics`.

File: deployment/environment_min.yml [hunk 3]
```diff
@@ -8,38 +8,38 @@ name: py37
...
-  - adaptive
-  - black
+  # qmt dependencies
   - dask
-  - dask-jobqueue
-  - deepdish
-  - descartes
-  - fenics
   - freecad
-  - future
```
Comment (basnijholt) [inline deployment/environment_min.yml]: this is not needed anymore because of Pytohn 3.7

**Comment by likewei92** [comment]
Nice thanks for the clean up

File: deployment/environment_min.yml [hunk 4]
```diff
@@ -8,38 +8,38 @@ name: py37
...
   - matplotlib
+  - scipy
+  - shapely
+  - sympy
+  # qms dependencies
+  - deepdish
+  - fenics
+  - kwant
   - mkl
-  - mkl-service
```
Comment (basnijholt) [inline deployment/environment_min.yml]: gets installed because of `mkl`.

File: deployment/environment_min.yml [hunk 5]
```diff
@@ -8,38 +8,38 @@ name: py37
...
-  - tqdm
+  # other tools (not needed for qmt/qms)
+  - adaptive
+  - dask-jobqueue
+  - hpc05
+  - jupyter
+  - jupyter_contrib_nbextensions
+  - mpi4py
+  - nb_conda
+  - rope
```
Comment (likewei92) [inline deployment/environment_min.yml]: I feel like rope should belong in the dev tools group, along with black, etc

Comment (basnijholt) [reply deployment/environment_min.yml]: But it's not used in the tests or CI.

Comment (likewei92) [reply deployment/environment_min.yml]: Ok sure. Let's keep it here then

**Pull request merged**